### PR TITLE
Setup CI

### PR DESCRIPTION
**Pull request by skshetry**
Part of issue #229 

- [x] Add linter
- [x] PHPunit tests
- [x] Some extra setups that might be required in the future

This PR doesnot fix any `formatting` errors, though there's the pipeline for `owncloud-coding-standard`, it is not enabled. I'll be making a issue for that.

### CODE REVIEW

**Comment by patrickjahns** [comment]
Please exclude:

- `vendor-bin/phpstan/composer.lock`
- `vendor-bin/owncloud-codestyle/composer.lock`
- `vendor-bin/phan/composer.lock` 

Reference `.gitignore` https://github.com/owncloud/password_policy/blob/master/.gitignore#L8

File: Makefile [hunk 1]
```diff
@@ -32,3 +32,139 @@ appstore:
...
+
+.PHONY: test-php-phpstan
+test-php-phpstan:          ## Run phpstan
+test-php-phpstan: vendor-bin/phpstan/vendor
+	$(PHPSTAN) analyse --memory-limit=4G --configuration=./phpstan.neon --no-progress --level=5 appinfo lib
+
+.PHONY: test-acceptance-api
+test-acceptance-api:       ## Run API acceptance tests
+test-acceptance-api: vendor/bin/phpunit
+	../../tests/acceptance/run.sh --type api
```
Comment (individual-it) [inline Makefile]: use  also `--remote`

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    when:
+      matrix:
+        TEST_SUITE: phpunit
+
+  codecov:
+    image: plugins/codecov:2
+    secrets: [codecov_token]
+    pull: true
+    paths:
+      - tests/output/clover.xml
```
Comment (individual-it) [inline .drone.yml]: put that into "files"

**Comment by individual-it** [comment]
this PR provides only the drone setup, there are no tests of any kind done for this app yet

**Comment by skshetry** [comment]
@ownclouders rebase

**Comment by skshetry** [comment]
@phil-davis @individual-it should we merge this?

**Pull request merged**